src/dmd/statementsem.d

```diff
@@ -1223,19 +1219,119 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
+        }
+
+        TypeAArray taa = null;
+        Type tn = null;
+        Type tnv = null;
+        Statement apply()
+        {
+            if (checkForArgTypes(fs))
+                return null;
+
+            TypeFunction tfld = null;
+            if (sapply)
+            {
+                FuncDeclaration fdapply = sapply.isFuncDeclaration();
+                if (fdapply)
+                {
+                    assert(fdapply.type && fdapply.type.ty == Tfunction);
+                    tfld = cast(TypeFunction)fdapply.type.typeSemantic(loc, sc2);
+                    goto Lget;
+                }
+                else if (tab.ty == Tdelegate)
+                {
+                    tfld = cast(TypeFunction)tab.nextOf();
+                Lget:
+                    //printf("tfld = %s\n", tfld.toChars());
+                    if (tfld.parameterList.parameters.dim == 1)
+                    {
+                        Parameter p = tfld.parameterList[0];
+                        if (p.type && p.type.ty == Tdelegate)
+                        {
+                            auto t = p.type.typeSemantic(loc, sc2);
+                            assert(t.ty == Tdelegate);
+                            tfld = cast(TypeFunction)t.nextOf();
+                        }
+                        //printf("tfld = %s\n", tfld.toChars());
+                    }
+                }
+            }
+
+            FuncExp flde = foreachBodyToFunction(sc2, fs, tfld);
+            if (!flde)
+                return null;
+
+            // Resolve any forward referenced goto's
+            foreach (ScopeStatement ss; *fs.gotos)
+            {
+                GotoStatement gs = ss.statement.isGotoStatement();
+                if (!gs.label.statement)
+                {
+                    // 'Promote' it to this scope, and replace with a return
+                    fs.cases.push(gs);
+                    ss.statement = new ReturnStatement(Loc.initial, new IntegerExp(fs.cases.dim + 1));
+                }
+            }
+
+            Expression e = null;
+            Expression ec;
+            if (vinit)
+            {
+                e = new DeclarationExp(loc, vinit);
+                e = e.expressionSemantic(sc2);
+                if (e.op == TOK.error)
+                    return null;
+            }
+
+            if (taa)
+                ec = applyAssocArray(fs, flde, taa);
+            else if (tab.ty == Tarray || tab.ty == Tsarray)
+                ec = applyArray(fs, flde, sc2, tn, tnv, tab.ty);
+            else if (tab.ty == Tdelegate)
+                ec = applyDelegate(fs, flde, sc2, tab);
+            else
+                ec = applyOpApply(fs, tab, sapply, sc2, flde);
+            if (!ec)
+                return null;
+            e = Expression.combine(e, ec);
+            return loopReturn(e, fs.cases, loc);
+        }
         switch (tab.ty)
         {
+        case Terror:
+            return retError();
+        default:
```

Comment:
default statement should do at the end of the case list.

Reply:
> default statement should go at the end of the case list.

done